core/arch/arm/encode.c

```diff
@@ -2766,7 +2766,7 @@ instr_encode_arch(dcontext_t *dcontext, instr_t *instr, byte *copy_pc, byte *fin
     }
 
     decode_info_init_for_instr(&di, instr);
-    di.opcode = instr_get_opcode(instr);
+    di.opcode = instr->opcode;
```

Comment:
I presume you mean: copy the 7 lines starting with "/* first handle the already-encoded instructions */" from x86/encode.c and insert them at line 2767. I'll test that tomorrow.